src/Types/BigIntType.php

```diff
@@ -47,15 +48,24 @@ public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int
             return $value;
         }
 
-        if ($value > PHP_INT_MIN && $value < PHP_INT_MAX) {
-            return (int) $value;
-        }
-
         assert(
             is_string($value),
             'DBAL assumes values outside of the integer range to be returned as string by the database driver.',
         );
 
+        $dotPos = strpos($value, '.');
+        if ($dotPos !== false && rtrim(substr($value, $dotPos + 1), '0') === '') {
```

Comment:
Psalm is broken - https://github.com/vimeo/psalm/issues/10995 - how to fix the CI?